src/Collection/Iterator/ZipIterator.php

```diff
@@ -71,33 +75,73 @@ class ZipIterator extends MultipleIterator implements CollectionInterface
      */
     public function __construct(array $sets, ?callable $callable = null)
     {
-        $sets = array_map(function ($items) {
-            return (new Collection($items))->unwrap();
-        }, $sets);
+        $this->multipleIterator = new MultipleIterator(
+            MultipleIterator::MIT_NEED_ALL | MultipleIterator::MIT_KEYS_NUMERIC
+        );
 
         $this->_callback = $callable;
-        parent::__construct(MultipleIterator::MIT_NEED_ALL | MultipleIterator::MIT_KEYS_NUMERIC);
 
         foreach ($sets as $set) {
-            $this->_iterators[] = $set;
-            $this->attachIterator($set);
+            $iterator = (new Collection($set))->unwrap();
+            $this->_iterators[] = $iterator;
+            $this->multipleIterator->attachIterator($iterator);
```

Comment:
Is it ok to assume Iterator here? The original code did the same but no stan error.

Reply:
For practical purposes it's safe to assume it will be an `Iterator`. We can later investigate if return type of `unwrap()` can be changed to `Iterator`.

Reply:
Changed the return type from `Traversable` to `Iterator` in #16089.

Reply:
Waiting for that PR to finish this one.